cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,504 @@
+	fs.StringVar(&manager.Master, "master", manager.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+	fs.StringVar(&manager.Kubeconfig, "kubeconfig", manager.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
+	fs.StringVar(&manager.pcrConfig, "pcrConfig", manager.pcrConfig, "Path to a single PCR config file")
+	fs.StringVar(&manager.pcrConfigDir, "pcrConfigDir", manager.pcrConfigDir, "Path to a PCR config directory")
+	fs.BoolVar(&manager.allowUnknown, "allowUnknown", false, "Allow unknown TPMs to join the cluster")
+	fs.IntVar(&manager.recurring, "reverify", 0, "Periodocally reverify nodes after this many seconds")
+}
+
+func updateConfig(configmap *api.ConfigMap) {
+	if configmap.Data["allowUnknown"] != "" {
+		allowUnknown, err := strconv.ParseBool(configmap.Data["allowUnknown"])
+		if err == nil {
+			manager.allowUnknown = allowUnknown
+		}
+	}
+	if configmap.Data["reverify"] != "" {
+		reverify, err := strconv.Atoi(configmap.Data["reverify"])
+		if err == nil {
+			manager.recurring = reverify
+			select {
+			case manager.recurringChan <- reverify:
+			default:
+			}
+		}
+	}
+}
+
+func run(stop <-chan struct{}) {
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: nodeAddFn,
+		},
+	)
+
+	_, configController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.ConfigMaps("kube-system").List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.ConfigMaps("kube-system").Watch(options)
+			},
+		},
+		&api.ConfigMap{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    configAddFn,
+			UpdateFunc: configUpdateFn,
+		},
+	)
+
+	_, policyController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.tpmhandler.PolicyClient.List(&options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.tpmhandler.PolicyClient.Watch(&options)
+			},
+		},
+		&runtime.Unstructured{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    policyAddFn,
+			UpdateFunc: policyUpdateFn,
+			DeleteFunc: policyDeleteFn,
+		},
+	)
+	fmt.Printf("Starting\n")
+	loadPolicy()
+	go reverify()
+	go nodeController.Run(wait.NeverStop)
+	go configController.Run(wait.NeverStop)
+	go policyController.Run(wait.NeverStop)
+	select {}
+	fmt.Printf("Returned\n")
```

Comment:
Unreachable code.  Is this function supposed to block on the stop channel and then proceed to tell the goroutines to exit? Or is this just kinda a "stopping to be implemented" kinda thing. If so, maybe just add a comment.